Integration/RESTTrait.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace WPMedia\PHPUnit\Integration;
+
+use WP_Rest_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+trait RESTTrait {
+
+	/**
+	 * Instance of the WordPress REST Server.
+	 *
+	 * @var WP_REST_Server
+	 */
+	protected $server;
+
+	/**
+	 * Sets up the WP REST API Server.
+	 */
+	protected function setUpServer() {
+		/**
+		 * @var WP_REST_Server $wp_rest_server
+		 */
+		global $wp_rest_server;
+		$this->server = $wp_rest_server = new WP_REST_Server;
+		do_action( 'rest_api_init' );
```

Comment:
Is there a reason not to use `rest_get_server()`?

```suggestion
		$this->server = rest_get_server();
```

Reply:
Hmm, originally I was thinking about bypassing the first filter to ensure we get `WP_Rest_Server`. But .... I think you're right.

Reply:
I was only asking actually, I'm not sure we should or shouldn't: both conflict with the precept "store previous state, work, restore stored state" in their current form.